### PR TITLE
fix(ext/node): better support for `node:diagnostics_channel` module

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/24060

### CODE REVIEW

File: ext/node/polyfills/diagnostics_channel.js [hunk 1]
```diff
@@ -0,0 +1,574 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
```
Comment (bartlomieju) [inline ext/node/polyfills/diagnostics_channel.js:2]: Note to self, add Node.js copyright as this is mostly copied from Node.js

**Review by littledivy — APPROVED** [review]
LGTM!

File: ext/node/polyfills/diagnostics_channel.js [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+function wrapStoreRun(store, data, next, transform = defaultTransform) {
+  return () => {
+    let context;
+    try {
+      context = transform(data);
+    } catch (err) {
+      nextTick(() => {
+        // TODO(bartlomieju): in Node.js this is using `triggerUncaughtException` API, need
+        // to clarify if we need that or if just throwing the error is enough here.
+        throw err;
```
Comment (devsnek) [inline ext/node/polyfills/diagnostics_channel.js:84]: i think `window.reportError` is mostly the equiv in deno? though i guess maybe this throw ends up there anyway?

**Review by devsnek — APPROVED** [review]
overall lgtm

File: ext/node/polyfills/diagnostics_channel.js (hunk 2 above)
Comment (bartlomieju) [reply ext/node/polyfills/diagnostics_channel.js:84]: Yeah, I believe it ends up there anyway. I'm gonna leave this TODO for now

**Pull request merged**